lib/Varien/File/Uploader/Image.php

```diff
@@ -26,7 +26,7 @@ class Varien_File_Uploader_Image extends Varien_File_Uploader
     public function __construct($file = null)
     {
         register_shutdown_function([$this, 'destruct']);
-        $this->newUploader($file);
+        $file = null;
```

Comment:
if we don't use a parameter phpstan will complain

Reply:
https://www.php.net/manual/en/class.iterator.php says the next() method returns void

Reply:
The interface says it returns void ... that method returned something.

Reply:
In this case since `$file` is not passed to parent class, I think it can be removed completely from the constructor.

Reply:
sure but I didn't want to change the signature because in theory it would be a breaking change. actually, for the next branch, this file can be removed because it's not used (as the treenode)

Reply:
I already tested it, it doesn't break if the argument is removed even if it's being passed explicitly to the constructor: https://3v4l.org/RDdcO

Reply:
sure sure, but the problem would be about somebody extending that class and getting a "method is not compatible with parent", am i right?

Reply:
Doesn't seem to be causing any issue either: https://3v4l.org/1PdNq, anyways I'm sleepy now so I might be missing something 😅

Reply:
@elidrissidev you're 100% right, I've changed it, thanks!